### PR TITLE
feat: add headers to UploadFile

### PR DESCRIPTION
**Pull request by adriangb**
Alternative / subset of #1311.

This preserves the multipart field headers for file uploads. Currently only `content_type` is preserved.

As per https://github.com/encode/starlette/pull/1311#issuecomment-999752023, this is pretty much the same API Flask uses.

### CODE REVIEW

File: starlette/datastructures.py [hunk 1]
```diff
@@ -415,15 +415,22 @@ class UploadFile:
...
+    headers: "Headers"
 
     def __init__(
-        self, filename: str, file: typing.IO = None, content_type: str = ""
+        self,
+        filename: str,
+        file: typing.IO = None,
+        content_type: str = "",
+        *,
+        raw_headers: typing.Optional[typing.List[typing.Tuple[bytes, bytes]]] = None,
```
Comment (lovelydinosaur) [inline starlette/datastructures.py]: I think it'd make more sense to me if the API here was `headers: Headers`.

The `formparsers.py` module could be responsible for instantiating the headers from the raw items, right?

**Comment by lovelydinosaur** [comment]
Yes looks good! Just the one comment here.

File: starlette/datastructures.py (hunk 1 above)
Comment (adriangb) [reply starlette/datastructures.py]: Yep great call, not sure why I wasn't doing that from the get go

**Comment by adriangb** [comment]
Addressed, thank you! I also added a blurb to the docs.

**Review by lovelydinosaur — APPROVED** [review]
Fantastic - thanks!

**Pull request merged**